src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -179,11 +200,37 @@ protected ExtensionsRunner(Extension extension) throws IOException {
+        Settings.Builder settingsBuilder = Settings.builder()
             .put(NODE_NAME_SETTING, extensionSettings.getExtensionName())
             .put(TransportSettings.BIND_HOST.getKey(), extensionSettings.getHostAddress())
-            .put(TransportSettings.PORT.getKey(), extensionSettings.getHostPort())
-            .build();
+            .put(TransportSettings.PORT.getKey(), extensionSettings.getHostPort());
+        boolean sslEnabled = extensionSettings.getOtherSettings().containsKey(SSL_TRANSPORT_ENABLED)
+            && "true".equals(extensionSettings.getOtherSettings().get(SSL_TRANSPORT_ENABLED));
+        if (sslEnabled) {
+            addSettingsToBuilder(settingsBuilder, "path.home", extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_ENABLED, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_PEMCERT_FILEPATH, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_PEMKEY_FILEPATH, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_PEMTRUSTEDCAS_FILEPATH, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_ENFORCE_HOSTNAME_VERIFICATION, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_ENFORCE_HOSTNAME_VERIFICATION_RESOLVE_HOST_NAME, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_EXTENDED_KEY_USAGE_ENABLED, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_CLIENT_PEMKEY_FILEPATH, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_CLIENT_PEMCERT_FILEPATH, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_CLIENT_PEMTRUSTEDCAS_FILEPATH, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_SERVER_PEMKEY_FILEPATH, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_SERVER_PEMCERT_FILEPATH, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_SERVER_PEMTRUSTEDCAS_FILEPATH, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_ENABLED_PROTOCOLS, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_ENABLED_CIPHERS, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_KEYSTORE_FILEPATH, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_KEYSTORE_ALIAS, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_KEYSTORE_TYPE, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_TRUSTSTORE_ALIAS, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_TRUSTSTORE_FILEPATH, extensionSettings);
+            addSettingsToBuilder(settingsBuilder, SSL_TRANSPORT_TRUSTSTORE_TYPE, extensionSettings);
```

Comment:
Probably have a utility function in `ExtensionSettings` and have a settingsBuilder?